query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -64,53 +64,56 @@ final case class InstantVectorFunctionMapper(function: InstantFunctionId,
-            // Special mapper to pull all buckets together from different Prom-schema time series
-            val mapper = HistogramQuantileMapper(funcParams)
-            mapper.apply(source, queryConfig, limit, sourceSchema, Observable.empty)
-          } else {
-            val instantFunction = InstantFunction.double(function)
-            source.map { rv =>
-              IteratorBackedRangeVector(rv.key, new DoubleInstantFuncIterator(rv.rows, instantFunction, scalarVector))
-            }
+        } else if (instantFunction.isHistDoubleToDoubleFunc && sourceSchema.isHistDouble) {
+          source.map { rv =>
+            IteratorBackedRangeVector(rv.key, new HD2DoubleInstantFuncIterator(rv.rows, instantFunction.asHDToDouble,
+              scalarRangeVector))
           }
-        case cType: ColumnType =>
-          throw new UnsupportedOperationException(s"Column type $cType is not supported for instant functions")
-      }
+        } else {
+          throw new UnsupportedOperationException(s"Sorry, function $function is not supported right now")
+        }
+      case ColumnType.DoubleColumn =>
+        if (function == HistogramQuantile) {
+          // Special mapper to pull all buckets together from different Prom-schema time series
+          val mapper = HistogramQuantileMapper(funcParams)
+          mapper.apply(source, queryConfig, limit, sourceSchema, Observable.empty)
+        } else {
+          val instantFunction = InstantFunction.double(function)
+          source.map { rv =>
+            IteratorBackedRangeVector(rv.key, new DoubleInstantFuncIterator(rv.rows, instantFunction,
+              scalarRangeVector))
+          }
+        }
+      case cType: ColumnType =>
+        throw new UnsupportedOperationException(s"Column type $cType is not supported for instant functions")
     }
+  }
 
+  def apply(source: Observable[RangeVector],
+            queryConfig: QueryConfig,
+            limit: Int,
+            sourceSchema: ResultSchema, paramResponse: Observable[ScalarRangeVector]): Observable[RangeVector] = {
     if (funcParams.isEmpty) {
-      evaluate(source, Nil)
+      evaluate(source, Nil, queryConfig, limit, sourceSchema)
     } else {
       funcParams.head match {
         case s: StaticFuncArgs   => evaluate(source, funcParams.map(x => x.asInstanceOf[StaticFuncArgs]).
-          map(x => ScalarFixedDouble(x.timeStepParams, x.scalar)))
+          map(x => ScalarFixedDouble(x.timeStepParams, x.scalar)), queryConfig, limit, sourceSchema)
         case t: TimeFuncArgs     => evaluate(source, funcParams.map(x => x.asInstanceOf[TimeFuncArgs]).
-          map(x => TimeScalar(x.timeStepParams)))
-        case e: ExecPlanFuncArgs => paramResponse.map(param => evaluate(source, Seq(param))).flatten
+          map(x => TimeScalar(x.timeStepParams)), queryConfig, limit, sourceSchema)
+        case e: ExecPlanFuncArgs => paramResponse.map(param => evaluate(source, Seq(param), queryConfig,
```

Comment:
Only use map(...) syntax if the closure is really small.  For this one refactor to

     paramResponse.map { param =>
        evalute(.....)
     }